src/main/kotlin/org/rust/lang/core/dfa/ControlFlowGraph.kt

```diff
@@ -338,8 +338,13 @@ private class ExitPointVisitor(
 
     private val RsExpr.isInTailPosition: Boolean
         get() {
+            // Ignore the expression if any of its ancestors are cfg-disabled
+            if (!existsAfterExpansion) return false
+
             for (ancestor in ancestors) {
                 when (ancestor) {
+                    // RsExprStmt (outside of RsTryExpr) may be RsExpr because of its right siblings' cfg attributes
+                    is RsExprStmt -> return ancestor.semicolon == null && ancestors.none { it is RsTryExpr }
```

Comment:
I'm wondering if an expression under cfg attribute could be something other than `RsBlockExpr`. If not, WDYT about using a more straightforward check, maybe like `ancestor is RsBlockExpr && ancestor.parent is RsExprStmt`? And if it could be something different, the corresponding tests can be added.

Reply:
BTW I find the entire `ExitPointVisitor` implementation quite hard to maintain and improve. I suppose it may take a long time to find a proper solution, even for simple problems. Perhaps we should redesign it at some point later